token-lending/program/src/processor.rs

```diff
@@ -1741,6 +1785,93 @@ fn process_liquidate_obligation(
+    let repay_reserve_liquidity_supply_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_collateral_mint_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_collateral_supply_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_liquidity_supply_info = next_account_info(account_info_iter)?;
+    let withdraw_reserve_liquidity_fee_receiver_info = next_account_info(account_info_iter)?;
+    let obligation_info = next_account_info(account_info_iter)?;
+    let lending_market_info = next_account_info(account_info_iter)?;
+    let lending_market_authority_info = next_account_info(account_info_iter)?;
+    let user_transfer_authority_info = next_account_info(account_info_iter)?;
+    let clock = &Clock::from_account_info(next_account_info(account_info_iter)?)?;
+    let token_program_id = next_account_info(account_info_iter)?;
+
+    let withdraw_collateral_amount = _liquidate_obligation(
+        program_id,
+        liquidity_amount,
+        source_liquidity_info,
+        destination_collateral_info,
+        repay_reserve_info,
+        repay_reserve_liquidity_supply_info,
+        withdraw_reserve_info,
+        withdraw_reserve_collateral_supply_info,
+        obligation_info,
+        lending_market_info,
+        lending_market_authority_info,
+        user_transfer_authority_info,
+        clock,
+        token_program_id,
+    )?;
+
+    _refresh_reserve_interest(program_id, withdraw_reserve_info, clock)?;
+    let withdraw_liquidity_amount = _redeem_reserve_collateral(
+        program_id,
+        withdraw_collateral_amount,
+        destination_collateral_info,
+        destination_liquidity_info,
+        withdraw_reserve_info,
+        withdraw_reserve_collateral_mint_info,
+        withdraw_reserve_liquidity_supply_info,
+        lending_market_info,
+        lending_market_authority_info,
+        user_transfer_authority_info,
+        clock,
+        token_program_id,
+    )?;
+    let mut withdraw_reserve = Reserve::unpack(&withdraw_reserve_info.data.borrow())?;
+    if &withdraw_reserve.config.fee_receiver != withdraw_reserve_liquidity_fee_receiver_info.key {
+        msg!("Withdraw reserve liquidity fee receiver does not match the reserve liquidity fee receiver provided");
+        return Err(LendingError::InvalidAccountInput.into());
+    }
+    let protocol_fee =
+        withdraw_reserve.calculate_protocol_liquidation_fee(withdraw_liquidity_amount)?;
+
+    spl_token_transfer(TokenTransferParams {
+        source: destination_liquidity_info.clone(),
+        destination: withdraw_reserve_liquidity_fee_receiver_info.clone(),
+        amount: protocol_fee,
+        authority: user_transfer_authority_info.clone(),
+        authority_signer_seeds: &[],
+        token_program: token_program_id.clone(),
+    })?;
+
+    withdraw_reserve.last_update.mark_stale();
+    Reserve::pack(
```

Comment:
Actually this might be wrong depending on now the account stuff actually works (See 2 comments above)

If the reserve that gets unpacked in this function scope doesn't include the changes caused by `redeem_reserve_collateral`, then the redeem reserve collateral changes will be lost I think